stories/uploads/components/single-image-upload-field.tsx

```
@@ -0,0 +1,319 @@
+  const inputId = useId();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [selectedFile, setSelectedFile] = useState<File>();
+  const [localPreview, setLocalPreview] = useState<string>();
+  const [progress, setProgress] = useState(0);
+  const [error, setError] = useState<string>();
+  const [isUploading, setIsUploading] = useState(false);
+  const [draftPosition, setDraftPosition] = useState<ImageObjectPosition>(
+    value?.objectPosition ?? DEFAULT_IMAGE_OBJECT_POSITION,
+  );
+  const draftPositionRef = useRef(draftPosition);
+  draftPositionRef.current = draftPosition;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/stories/uploads/components/single-image-upload-field.tsx:89:3
  87 |   );
  88 |   const draftPositionRef = useRef(draftPosition);
> 89 |   draftPositionRef.current = draftPosition;
     |   ^^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  90 |
  91 |   // Reset the draft crop when a different uploaded image is set, not when the
  92 |   // user pans the current one (that only changes objectPosition).

(react-hooks/refs)